worker-sys/src/types/socket.rs

```diff
@@ -0,0 +1,121 @@
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen(module = "cloudflare:sockets")]
+extern "C" {
+    #[wasm_bindgen]
+    pub fn connect(address: JsValue, options: JsValue) -> Socket;
+}
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(extends=js_sys::Object)]
+    #[derive(Debug, Clone, PartialEq, Eq)]
+    pub type Socket;
+
+    #[wasm_bindgen(method)]
+    pub fn close(this: &Socket) -> js_sys::Promise;
+
+    #[wasm_bindgen(method)]
+    pub fn closed(this: &Socket) -> js_sys::Promise;
+
+    #[wasm_bindgen(method, js_name=startTls)]
+    pub fn start_tls(this: &Socket) -> Socket;
+}
+
+impl Socket {
+    pub fn readable(&self) -> ReadableStream {
+        js_sys::Reflect::get(self, &JsValue::from("readable"))
+            .unwrap()
+            .into()
+    }
+
+    pub fn writable(&self) -> WritableStream {
+        js_sys::Reflect::get(self, &JsValue::from("writable"))
+            .unwrap()
+            .into()
+    }
+}
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(extends=js_sys::Object)]
+    #[derive(Debug, Clone, PartialEq, Eq)]
+    pub type ReadableStream;
```

Comment:
We shouldn't redefine this and instead use [ReadableStream](https://docs.rs/web-sys/latest/web_sys/struct.ReadableStream.html) from the existing web-sys crate.